src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Converters/CodeOfTToTokenSearchValueTypeConverter.cs

```diff
@@ -31,9 +31,16 @@ public IEnumerable<ISearchValue> ConvertTo(object value)
 
             EnsureArg.IsTrue(type.IsGenericType && type.GetGenericTypeDefinition() == FhirElementType, nameof(value));
 
-            ISystemAndCode systemAndCode = (ISystemAndCode)value;
+            var systemAndCode = value as ISystemAndCode;
 
-            yield return new TokenSearchValue(systemAndCode.System, systemAndCode.Code, null);
+            if (systemAndCode == null || systemAndCode.Code == null)
```

Comment:
Should `systemAndCode.Code` be exactly null or can IsNullOrWhitespace also count?

Reply:
Also is there a case where we System would be defined and not code?
Is the example from https://www.hl7.org/fhir/datatypes-examples.html#codeableconcept valid here:
```xml
 <unit>
    <coding>
      <system value="urn:oid:2.16.840.1.113883.19.5.2" />
      <code value="tab" />
      <display value="Tablet" />
    </coding>
    <coding>
      <system value="http://unitsofmeasure.org" />
    </coding>
  </unit>
```

Reply:
I took a quick look at the firely code, where the exception is originating (https://github.com/FirelyTeam/fhir-net-api/blob/add8073b8d096f66cde13f54c940048ba4363e0a/src/Hl7.Fhir.Core/Model/Code.cs) and it seems we will only get the exception if the code is null.  If the System is null, we won't see an exception, we could create a search param with an empty system.  Currently our TokenSearchParam class allows that.  But perhaps we should not?